Storage/docu/docu_3_code.cc

```diff
@@ -403,4 +413,82 @@ TEST(StorageDocumentation, RESTifiedStorageExample) {
 
 }
 
+namespace storage_docu {
+
+// Example of custom REST implementation that annotates transactions.
+struct RESTWithMeta: current::storage::rest::AdvancedHypermedia {
+  using SUPER = current::storage::rest::AdvancedHypermedia;
+
+  template <class HTTP_VERB, typename PARTICULAR_FIELD, typename ENTRY, typename KEY>
+  struct RESTful: SUPER::RESTful<HTTP_VERB, PARTICULAR_FIELD, ENTRY, KEY> {
+    using ACTUAL_SUPER = SUPER::RESTful<HTTP_VERB, PARTICULAR_FIELD, ENTRY, KEY>;
+
+    template <class INPUT>
+    Response Run(const INPUT& input) const {
+      input.fields.SetTransactionMetaField("who", "unittest");
+      return this->ACTUAL_SUPER::template Run<INPUT>(input);
+    }
+  };
+};
+
+}  // namespace storage_docu
+
+TEST(StorageDocumentation, RESTFillingTransactionMetaExample) {
+  using namespace current::storage::rest;
+  using namespace storage_docu;
+  using TestStorage = StorageOfClients<SherlockStreamPersister>;
+
+  const std::string client_storage_file_name =
+      current::FileSystem::JoinPath(FLAGS_client_storage_test_tmpdir, "client_with_meta");
+  const auto client_storage_file_remover = current::FileSystem::ScopedRmFile(client_storage_file_name);
+  TestStorage storage(client_storage_file_name);
+
+  const auto rest = RESTfulStorage<TestStorage, RESTWithMeta>(
+      storage,
+      FLAGS_client_storage_test_port,
+      "/api",
+      "http://example.current.ai/api");
+  const auto base_url = current::strings::Printf("http://localhost:%d", FLAGS_client_storage_test_port);
+
+  // Add client.
+  current::time::SetNow(std::chrono::microseconds(1024));
+  EXPECT_EQ(201, static_cast<int>(HTTP(PUT(base_url + "/api/data/client/101", Client(ClientID(101)))).code));
+  // Delete client.
+  current::time::SetNow(std::chrono::microseconds(2000));
+  EXPECT_EQ(200, static_cast<int>(HTTP(DELETE(base_url + "/api/data/client/101")).code));
+
+  // Check that everything has been persisted correctly, including meta fields.
+  const auto persisted_entries =
+      current::strings::Split<current::strings::ByLines>(current::FileSystem::ReadFileAsString(client_storage_file_name));
+  ASSERT_EQ(2u, persisted_entries.size());
+
+  const std::string& persisted_add = persisted_entries[0];
+  const size_t tab_pos = persisted_add.find('\t');
```

Comment:
`const auto add_fields = strings::Split(persisted_add, '\t'); ASSERT_TRUE(add_fields.size() == 2u);` is probably cleaner.